numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -4111,37 +4099,32 @@ initialize_numeric_types(void)
     /**end repeat**/
 
 
-    /*
-     * These need to be coded specially because getitem does not
-     * return a normal Python type
-     */
-    PyLongDoubleArrType_Type.tp_as_number = &longdoubletype_as_number;
-    PyCLongDoubleArrType_Type.tp_as_number = &clongdoubletype_as_number;
 
     /**begin repeat
-     * #name = int, float, repr, str#
-     * #kind = tp_as_number->nb*2, tp*2#
+     * #char = ,c#
+     * #CHAR = ,C#
      */
 
-    PyLongDoubleArrType_Type.@kind@_@name@ = longdoubletype_@name@;
-    PyCLongDoubleArrType_Type.@kind@_@name@ = clongdoubletype_@name@;
-
-    /**end repeat**/
-
-
-#if !defined(NPY_PY3K)
-    /**begin repeat
-     * #name = long, hex, oct#
-     * #kind = tp_as_number->nb*3#
+    /*
+     * These need to be coded specially because longdouble/clongdouble getitem
+     * does not return a normal Python type
      */
+    @char@longdoubletype_as_number.nb_float = @char@longdoubletype_float;
+#if defined(NPY_PY3K)
+    @char@longdoubletype_as_number.nb_int  = @char@longdoubletype_long;
+#else
+    @char@longdoubletype_as_number.nb_int  = @char@longdoubletype_int;
+    @char@longdoubletype_as_number.nb_long = @char@longdoubletype_long;
+    @char@longdoubletype_as_number.nb_hex  = @char@longdoubletype_hex;
+    @char@longdoubletype_as_number.nb_oct  = @char@longdoubletype_oct;
+#endif
 
-    PyLongDoubleArrType_Type.@kind@_@name@ = longdoubletype_@name@;
-    PyCLongDoubleArrType_Type.@kind@_@name@ = clongdoubletype_@name@;
+    Py@CHAR@LongDoubleArrType_Type.tp_as_number = &@char@longdoubletype_as_number;
+    Py@CHAR@LongDoubleArrType_Type.tp_repr = @char@longdoubletype_repr;
+    Py@CHAR@LongDoubleArrType_Type.tp_str = @char@longdoubletype_str;
```

Comment:
It's far more readable and concise to iterate over the types here, rather than the members.